src/phoenix/db/models.py

```diff
@@ -616,3 +616,65 @@ class ExperimentRunAnnotation(Base):
             "name",
         ),
     )
+
+
+class UserRole(Base):
+    __tablename__ = "user_roles"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    role: Mapped[str] = mapped_column(
+        CheckConstraint("role IN ('SYSTEM', 'ADMIN', 'GENERAL')", name="valid_role"),
+        unique=True,
+    )
+
+
+class User(Base):
+    __tablename__ = "users"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_role_id: Mapped[int] = mapped_column(
+        ForeignKey("user_roles.id"),
+        index=True,
+    )
+    username: Mapped[Optional[str]] = mapped_column(nullable=True, unique=True, index=True)
+    email_address: Mapped[Optional[str]] = mapped_column(nullable=True, unique=True, index=True)
+    auth_method: Mapped[str] = mapped_column(
+        CheckConstraint("auth_method IN ('LOCAL')", name="valid_auth_method")
+    )
+    password_hash: Mapped[Optional[str]]
+    reset_password: Mapped[bool]
+    created_at: Mapped[datetime] = mapped_column(UtcTimeStamp, server_default=func.now())
+    updated_at: Mapped[datetime] = mapped_column(
+        UtcTimeStamp, server_default=func.now(), onupdate=func.now()
+    )
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(UtcTimeStamp)
+
+
+class APIKey(Base):
+    __tablename__ = "api_keys"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(
+        ForeignKey("users.id"),
+        index=True,
+    )
+    name: Mapped[str]
+    description: Mapped[Optional[str]]
+    created_at: Mapped[datetime] = mapped_column(UtcTimeStamp, server_default=func.now())
+    expires_at: Mapped[Optional[datetime]] = mapped_column(UtcTimeStamp)
+
+
+class APIKeysAction(Base):
+    __tablename__ = "api_keys_actions"
```

Comment:
I would name this audit_api_keys so we can just make a consistent table - something like https://www.red-gate.com/simple-talk/databases/sql-server/database-administration-sql-server/pop-rivetts-sql-server-faq-no-5-pop-on-the-audit-trail/

Reply:
will cover in a follow up https://github.com/Arize-ai/phoenix/pull/4113